docs/cpp/destructors-cpp.md

```diff
@@ -71,7 +71,7 @@ Destructors are called when one of the following events occurs:
 
 - A local (automatic) object with block scope goes out of scope.
 
-- An object allocated using the **`new`** operator is explicitly deallocated using **`delete`**.
+- An object allocated using the **`new`** or **`new[]`** operator is explicitly deallocated using **`delete`** or **`delete[]`**.
```

Comment:
I see the parallelism here, but let's make it more explicit since you shouldn't use delete to deallocate memory allocated with new[]. Something like "An object allocated using **`new`** should be deallocated with **`delete`**. An object allocated using **`new[]`** should be deallocated with **`delete[]`**

Reply:
I can't think of a way to make it flow well if we keep to 1 bullet point, hence, I split it into 2. I kept the phrasing of the first part of each bullet point, since it is in the `Destructors are called when one of the following events occurs:` part.

Reply:
I took a whack at it.